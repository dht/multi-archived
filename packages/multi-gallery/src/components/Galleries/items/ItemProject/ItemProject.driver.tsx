import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ItemProject, ItemProjectProps } from './ItemProject';
import { BaseComponentDriver } from 'testing-base';

export class ItemProjectDriver extends BaseComponentDriver {
    private props: Partial<ItemProjectProps> = {};

    constructor() {
        super('ItemProject');
    }

    when: any = {
        rendered: () => {
            render(<ItemProject {...(this.props as ItemProjectProps)} />);
            return this;
        },
        clicked: () => {
            fireEvent.click(this.wrapper);
            return this;
        },
        snapshot: () => {
            return this.snapshot(
                <ItemProject {...(this.props as ItemProjectProps)} />
            );
        },
    };

    given: any = {
        props: (props: Partial<ItemProjectProps>) => {
            this.props = props;
            return this;
        },
    };

    get = {
        containerClassName: () => {
            return this.wrapper.className;
        },
        label: () => {
            return this.wrapper.innerHTML;
        },
    };
}
