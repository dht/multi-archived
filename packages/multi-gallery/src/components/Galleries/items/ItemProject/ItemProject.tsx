import React from 'react';
import { Wrapper } from './ItemProject.style';

export type ItemProjectProps = {};

export function ItemProject(_props: ItemProjectProps) {
    return (
        <Wrapper
            className='ItemProject-wrapper'
            data-testid='ItemProject-wrapper'
        >
            ItemProject
        </Wrapper>
    );
}

export default ItemProject;
