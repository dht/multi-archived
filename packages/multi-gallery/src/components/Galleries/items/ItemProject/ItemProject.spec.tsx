import { ItemProjectDriver } from './ItemProject.driver';
import Chance from 'chance';

const chance = new Chance();

describe('ItemProject', () => {
    let driver: ItemProjectDriver;

    beforeAll(() => {
        driver = new ItemProjectDriver();
    });

    it('should render button', () => {
        const label = chance.word();

        const element = driver.given
            .props({
                title: label,
            })
            .when.rendered();

        const containerClassName = element.get.containerClassName();
        const innerText = element.get.label();

        expect(containerClassName).toContain('ItemProject-wrapper');
        expect(innerText).toBe(label);
    });

    it('should click button', () => {
        const callback = jest.fn();

        driver.given
            .props({
                onClick: callback,
            })
            .when.rendered()
            .when.clicked();

        expect(callback).toHaveBeenCalledTimes(1);
    });
});

describe('ItemProject snapshots', () => {
    let driver: ItemProjectDriver;

    beforeAll(() => {
        driver = new ItemProjectDriver();
    });

    it('should match snapshot', () => {
        expect(driver.when.snapshot()).toMatchSnapshot();
    });
});
