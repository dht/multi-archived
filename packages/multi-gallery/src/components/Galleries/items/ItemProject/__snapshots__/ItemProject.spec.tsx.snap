// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemProject snapshots should match snapshot 1`] = `
<div
  className="sc-bczRLJ WtYQR ItemProject-wrapper"
  data-testid="ItemProject-wrapper"
>
  ItemProject
</div>
`;
